fix: LEAP-930: Changes to ML settings UI

hlomzik opened the pull request:
Annotation settings:
- remove links to docs as they don't fully describe the flow

Model settings:
- also remove links
- fix text describing how to use new model in labeling
- add link to Annotation settings
- simplify blocks used on a page

<img width="456" alt="Screenshot 2024-04-10 at 23 58 00" src="https://github.com/HumanSignal/label-studio/assets/1607227/fc0084e2-0f26-459e-b282-6abf21a00b92">
<img width="688" alt="Screenshot 2024-04-10 at 23 57 48" src="https://github.com/HumanSignal/label-studio/assets/1607227/7a79402b-ebd9-40cb-837a-cc2c038a8b42">


Mirroring #5696 for Follow Merge

### PR fulfills these requirements
- [ ] Tests for the changes have been added/updated (for bug fixes/features)
- [ ] Docs have been added/updated (for bug fixes/features)
- [x] Best efforts were made to ensure docs/code are concise and coherent (checked for spelling/grammatical errors, commented out code, debug logs etc.)
- [x] Self-reviewed and ran all changes on a local instance (for bug fixes/features)

#### Change has impacts in these area(s)
- [ ] Product design
- [ ] Backend (Database)
- [ ] Backend (API)
- [x] Frontend

Commit:
changing the copy back and providing more info

Commit:
changing the copy

Commit:
fixing structure

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8611807449

deppp committed:
Update web/apps/labelstudio/src/components/Description/Description.styl

Co-authored-by: hlomzik <hlomzik@gmail.com>

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8619449452

hlomzik committed:
Fix branch name

hlomzik committed:
Improve style and mock-ups

hlomzik committed:
Simplify callout block in Model settings

We autoselect model_version now, so user action is not required.

yyassi-heartex commented on web/apps/labelstudio/src/pages/Settings/MachineLearningSettings/MachineLearningSettings.js:
```diff
@@ -132,33 +132,24 @@ export const MachineLearningSettings = () => {
-          </Description>
+          <>
+            <Description>
+              A connected model has been detected! If you wish to fetch
+              predictions from this model, please follow these steps:
+              <br />
+              <br />
+              1. Navigate to the <i>Data Manager</i>.<br />
+              2. Select the desired tasks.<br />
+              3. Click on <i>Retrieve predictions</i> from the <i>Actions</i> menu.
```
I see this is copy/paste from before but why are we relying on `<br />` for spacing and layout? also this seems like an optimal spot to use `<ol>` for the list - can we get a follow up ticket to address this?

hlomzik replied:
I'm going to fix this asap, yes, thank you!

yyassi-heartex reviewed (approved):
approving with the expectation that there will be a follow up ticket to address my concern regarding the layout/component structure of the description content

hlomzik commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/8741386553)
> Successfully merged:  441 files changed, 2628 insertions(+), 2411 deletions(-)

hlomzik committed:
Merge branch 'develop' into 'fb-leap-930/copy-changes'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8741386553

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8741408087
